go/types/value_store.go

```diff
@@ -284,14 +303,29 @@ func (lvs *ValueStore) bufferChunk(v Value, c chunks.Chunk, height uint64) {
 	}
 }
 
+func (lvs *ValueStore) Root() hash.Hash {
+	return lvs.cs.Root()
+}
+
+func (lvs *ValueStore) Rebase() {
+	lvs.cs.Rebase()
+}
+
 // Flush() puts all bufferedChunks into the ChunkStore, with best-effort
 // locality. NB: The Chunks will not be made durable unless the caller also
 // Commits to the underlying ChunkStore.
-func (lvs *ValueStore) Flush() {
-	func() {
+func (lvs *ValueStore) Commit(current hash.Hash) bool {
+	return func() bool {
 		lvs.bufferMu.Lock()
 		defer lvs.bufferMu.Unlock()
 
+		checkCurrent := false
```

Comment:
it's a little weird to frontload this check when you don't need to know this until way down below. Indeed, you only need to know this if you're checking completeness.